### PR TITLE
Update helm integration tests

### PR DESCRIPTION
**Pull request by jacob-delgado**
**Please provide a description of this PR:**

Update the helm tests as part of helm graduation from Alpha to Beta. (https://github.com/istio/istio/issues/42565)

1. Replace old gateway charts with the gateway chart
2. Remove egress gateway chart from install
3. Update n-2 to Istio 1.15.0
4. Update n-1 test to Istio 1.16.0
5. Remove older versions
6. Remove restriction on the max k8s version being 1.22
7. Refer to tests based on their usage in docs and not in the helm values

### CODE REVIEW

**Comment by jacob-delgado** [comment]
Once this gets approved I can start working on using the helm repo to do an install alongside the tarballs.

We have users using both the tarballs as well as the helm repo. We should nudge these users onto the helm repo as the only officially supported helm installation method.

File: tests/integration/helm/upgrade/helm_upgrade_test.go [hunk 1]
```diff
@@ -18,16 +18,14 @@
...
-	"strings"
 	"testing"
 
 	"istio.io/istio/pkg/test/framework"
 )
 
 const (
-	previousSupportedVersion = "1.11.3"
-	nMinusTwoVersion         = "1.10.0"
+	previousSupportedVersion = "1.16.0"
```
Comment (kfaseela) [inline tests/integration/helm/upgrade/helm_upgrade_test.go:27]: is there a way to get rid of this hardcoding? and automatically take the latest version and a Minus Version?

File: tests/integration/helm/upgrade/helm_upgrade_test.go [hunk 2]
```diff
@@ -18,16 +18,14 @@
...
+	nMinusTwoVersion         = "1.15.0"
```
Comment (kfaseela) [inline tests/integration/helm/upgrade/helm_upgrade_test.go:28]: if Minus Two, we need 1.14? or else change the variable name?

Comment (jacob-delgado) [reply tests/integration/helm/upgrade/helm_upgrade_test.go:28]: It's 2 from the branch. So it'd be 1.15 as master branch points to 1.17.

File: tests/integration/helm/upgrade/helm_upgrade_test.go (hunk 1 above)
Comment (jacob-delgado) [reply tests/integration/helm/upgrade/helm_upgrade_test.go:27]: I think there's a 1.17-dev in the makefile. I can see if there's a variable in a later or. I'm planning to clean this up more but didn't want to do it until it's up to date.

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
Helm repo adds a external dependency to tests, may not be desired (https://github.com/istio/istio/issues/35835)

**Comment by nshankar13** [comment]
@howardjohn @kfaseela the [integration test guide](https://github.com/istio/istio/tree/master/tests/integration#test-selection) explains how to skip tests based on the "label," but what about tests specified as "Features" such as the first-party JWT test? https://github.com/istio/istio/blob/master/tests/integration/helm/install_test.go#L59

**Comment by howardjohn** [comment]
There is no option for that. I would just skip by name personally. There is `-istio.test.skip` (which was added to Go natively I think, so probably you can use the Go flag as well)

**Comment by nshankar13** [comment]
@howardjohn `istio.test.skip` expects the goTest.name I think. Would it look something like this? `--istio.test.skip="\binstallation.helm.firstpartyjwt.install\b"`

**Comment by howardjohn** [comment]
No, you use the go test name. No support for skipping features, use name instead

**Pull request merged**